syntax = "proto3";
package canlink;
message RawFrame {
  double  timestamp        = 1;
  uint32  can_id           = 2;
  bool    is_extended_id   = 3;
  bytes   data             = 4;
  uint32  dlc              = 5;
}
message FrameBatch {
  repeated RawFrame frames = 1;
}

message UploadAck {
  uint64 frames_ingested = 1;
}

service CanIngest {
  rpc UploadFrames (stream FrameBatch) returns (UploadAck);
}
